utils/resolve.js

```diff
@@ -80,19 +80,25 @@ exports.fileExistsWithCaseSync = function fileExistsWithCaseSync(filepath, cache
   return result;
 };
 
-function relative(modulePath, sourceFile, settings) {
-  return fullResolve(modulePath, sourceFile, settings).path;
+function relative(modulePath, sourceFile, context) {
+  return fullResolve(
+    modulePath,
+    sourceFile,
+    // Accept context or settings, for backward compatibility
+    !('settings' in context) ? { settings: context } : context,
```

Comment:
I’m pretty skeptical there’s always a “settings” in context, consistently from eslint 2 - 7.

Reply:
Well, we pass `context.settings` -> it today, so we already rely on that, don't we?
https://github.com/import-js/eslint-plugin-import/blob/dd814245b7f769a84235f2ffb7204287d47001ba/utils/resolve.js#L220
https://github.com/import-js/eslint-plugin-import/blob/dd814245b7f769a84235f2ffb7204287d47001ba/src/ExportMap.js#L434

Reply:
i guess that's true; if it wasn't an object it'd throw since it's not checked for truthiness. Still, is there a more robust way to check? perhaps eslint provides a Context constructor or something we could use to compare it to?

Reply:
Good question ... Here's where it's constructed today, I think: https://github.com/eslint/eslint/blob/cf34e5cf5ed5d09eb53c16cca06821c4e34b7b70/lib/linter/linter.js#L910-L954

Looking at that, I'm not sure what would be more robust, and consistent between ESLint versions?

Reply:
Another alternative is to always pass `settings`, and just *also* pass `context`?

Reply:
Totally -- I propose overloading the argument because I think it's functionally equivalent, and slightly tidier than calling `relative(modulePath, sourceFile, context.settings, context)`, but no strong feelings either way ...

Reply:
it's definitely tidier - we could also allow `relative(modulePath, sourceFile, null, context)`, and adapt it later whenever we do a semver-major